### PR TITLE
Fix issue where the select does not always handle a change in the content array properly

### PR DESCRIPTION
**Pull request by misquith**
This is a minor change to the dependent properties of groupedContent so that it mirrors the dependent property list of preparedContent. Specifically that dependent property depends on the `property` as well as `property.[]`

Use case:

The content attribute is set an empty array and then immediately to a non empty array. 

Bug: The select field does not show any content

### CODE REVIEW

**Comment by cyril-sf** [comment]
can you write a failing test?

File: addon/components/select-component.js [hunk 1]
```diff
@@ -293,7 +296,7 @@ export default Ember.Component.extend(
       return result.pushObjects(groups[key]);
     });
     return result;
-  }).property('preparedContent.[]', 'optionGroupPath', 'labels.[]'),
+  }).property('preparedContent', 'preparedContent.[]', 'optionGroupPath', 'labels.[]'),
```
Comment (cyril-sf) [inline addon/components/select-component.js]: There's probably no reason to have `'preparedContent` and `'preparedContent.[]`

**Comment by misquith** [comment]
@cyril-sf Added a test which fails if preparedContent is not specified as a dependent property on groupedContent. Please take a look

**Comment by misquith** [comment]
bump

**Comment by misquith** [comment]
bump

File: addon/components/select-component.js [hunk 2]
```diff
@@ -345,8 +349,7 @@ export default Ember.Component.extend(
       result.pushObjects(groupObj.members);
     });
     return result;
-  }).property('preparedContent.[]', 'optionGroupPath', 'labels.[]',
-    'groupSortFunction'),
+  }).property('preparedContent', 'preparedContent.[]', 'optionGroupPath', 'labels.[]', 'groupSortFunction'),
```
Comment (cyril-sf) [inline addon/components/select-component.js]: You don't need `'preparedContent'` and `'preparedContent.[]'` The latter should be enough

File: tests/integration/select-component-test.js [hunk 3]
```diff
@@ -444,3 +444,24 @@ test('Can specify a custom partial with listViewPartial', function(assert) {
...
+  expect(2);
+
+  var resultItemSelector = '.ember-select-result-item';
+  select = this.subject({
+    content: ['test content']
+  });
+  this.append();
+  var selectElement = select.$();
+  openDropdown(selectElement);
+  wait();
```
Comment (cyril-sf) [inline tests/integration/select-component-test.js]: this should not be needed

File: tests/integration/select-component-test.js [hunk 4]
```diff
@@ -444,3 +444,24 @@ test('Can specify a custom partial with listViewPartial', function(assert) {
       'The specified listViewPartial is rendered');
   });
 });
+
+test('Displays options when content is changed to another empty array and then immediately assigned to a content array', function() {
```
Comment (cyril-sf) [inline tests/integration/select-component-test.js]: Not sure this describes well what the test want to capture. 

You want something like

```
Select handles a change in the content array properly
```

File: addon/components/select-component.js (hunk 2 above)
Comment (misquith) [inline addon/components/select-component.js]: The test in the PR fails if 'preparedContent' is not specified

**Pull request merged**